### PR TITLE
Cleanup `Promise.resolve`s

### PR DESCRIPTION
**Pull request by josh**
Silly nitpick I caught while reviewing the extension JS.

You can kinda pretend all return values within a `then` are automatically wrapped in `Promise.resolve`.

These two statements return the same promised wrapped `42`:
- `Promise.resolve(42)`
- `Promise.resolve().then(() => { return 42 })`

You can also return Promise's at the end of `then()`, and it will be "flattened". So returning a type `Promise` has a different effect than any other value.
- `Promise.resolve(42)`
- `Promise.resolve().then(() => { return Promise.resolve(Promise.resolve(42)) }).then(value => Promise.resolve(value))`.

Second, instead of `reject`, you can simply `throw` inside a `then()`.

To sum up, inside a `then()` block, you shouldn't need to use any `Promise.***` helper.
## 

To: @mastahyeti 
CC: @dgraham

### CODE REVIEW

**Comment by btoews** [comment]
Thanks @josh! `Action.js` is the result of "compiling" all the stuff from `/JavaScript/common/`, `/JavaScript/extension/` and `/JavaScript/extension.js` using a custom XCode "build phase" ([script](https://github.com/mastahyeti/security-key/blob/master/SecurityKey.xcodeproj/project.pbxproj#L262)). I'll copy these changes over to the source files and recompile things.

**Comment by josh** [comment]
> Action.js is the result of "compiling" all the stuff from /JavaScript/common/, /JavaScript/extension/ and /JavaScript/extension.js using a custom XCode "build phase" (script). I'll copy these changes over to the source files and recompile things.

D'oh! my bad. Hopefully this at least makes sense.

**Comment by btoews** [comment]
> Hopefully this at least makes sense.

Definitely. Much appreciated :bow:
